tests/quantization/finegrained_fp8/test_finegrained_fp8.py

```diff
@@ -0,0 +1,627 @@
+        """
+        model_size = 0
+        for name, child in model.named_children():
+            if not (isinstance(child, torch.nn.Embedding) and ignore_embeddings):
+                for p in child.parameters(recurse=False):
+                    model_size += p.numel() * p.element_size()
+                for b in child.buffers(recurse=False):
+                    model_size += b.numel() * b.element_size()
+                model_size += self.get_model_size_in_bytes(child, ignore_embeddings)
+        return model_size
+    
+    def get_dummy_inputs(self, device: torch.device, seed: int = 0):
+        if str(device).startswith("mps"):
+            generator = torch.manual_seed(seed)
+        else:
+            generator = torch.Generator().manual_seed(seed)
+
+        inputs = {
+            "prompt": "an astronaut riding a horse in space",
+            "height": 32,
+            "width": 32,
+            "num_inference_steps": 2,
+            "output_type": "np",
+            "generator": generator,
+        }
+
+        return inputs
+
+    def get_dummy_tensor_inputs(self, device=None, seed: int = 0):
+        batch_size = 1
+        num_latent_channels = 4
+        num_image_channels = 3
+        height = width = 4
+        sequence_length = 48
+        embedding_dim = 32
+
+        torch.manual_seed(seed)
+        hidden_states = torch.randn((batch_size, height * width, num_latent_channels)).to(device, dtype=torch.bfloat16)
+
+        torch.manual_seed(seed)
+        encoder_hidden_states = torch.randn((batch_size, sequence_length, embedding_dim)).to(
+            device, dtype=torch.bfloat16
+        )
+
+        torch.manual_seed(seed)
+        pooled_prompt_embeds = torch.randn((batch_size, embedding_dim)).to(device, dtype=torch.bfloat16)
+
+        torch.manual_seed(seed)
+        text_ids = torch.randn((sequence_length, num_image_channels)).to(device, dtype=torch.bfloat16)
+
+        torch.manual_seed(seed)
+        image_ids = torch.randn((height * width, num_image_channels)).to(device, dtype=torch.bfloat16)
+
+        timestep = torch.tensor([1.0]).to(device, dtype=torch.bfloat16).expand(batch_size)
+
+        return {
+            "hidden_states": hidden_states,
+            "encoder_hidden_states": encoder_hidden_states,
+            "pooled_projections": pooled_prompt_embeds,
+            "txt_ids": text_ids,
+            "img_ids": image_ids,
+            "timestep": timestep,
+        }
+
+    def _test_quantization_output(self, quantization_config: FinegrainedFP8Config, expected_slice: List[float], model_id: str):
+        components = self.get_dummy_components(quantization_config, model_id)
+        pipe = FluxPipeline(**components)
+        pipe.to(device=torch_device)
+
+        inputs = self.get_dummy_inputs(torch_device)
+        output = pipe(**inputs)[0]
+        output_slice = output[-1, -1, -3:, -3:].flatten()
+
+        self.assertTrue(np.allclose(output_slice, expected_slice, atol=1e-3, rtol=1e-3))
```

Comment:
Could we use cosine similarity instead?
https://github.com/huggingface/diffusers/blob/f0dba33d82af991369806312f61ab4c6cb7a8dd1/tests/quantization/bnb/test_4bit.py#L476